samples/server/petstore/kotlin-server/javalin/src/main/kotlin/org/openapitools/server/Main.kt

```diff
@@ -0,0 +1,48 @@
+package org.openapitools.server
+
+import io.javalin.Javalin
+import io.javalin.apibuilder.ApiBuilder.*
+
+import org.openapitools.server.apis.PetApi
+import org.openapitools.server.apis.PetApiServiceImpl
+import org.openapitools.server.apis.StoreApi
+import org.openapitools.server.apis.StoreApiServiceImpl
+import org.openapitools.server.apis.UserApi
+import org.openapitools.server.apis.UserApiServiceImpl
+
+fun main() {
+    val PetApi = PetApi(PetApiServiceImpl())
+    val StoreApi = StoreApi(StoreApiServiceImpl())
+    val UserApi = UserApi(UserApiServiceImpl())
+
+    val app = Javalin
+        .create()
+        .routes {
+            post("/pet", PetApi::addPet)
+            delete("/pet/{petId}", PetApi::deletePet)
+            get("/pet/findByStatus", PetApi::findPetsByStatus)
+            get("/pet/findByTags", PetApi::findPetsByTags)
+            get("/pet/{petId}", PetApi::getPetById)
+            put("/pet", PetApi::updatePet)
+            post("/pet/{petId}", PetApi::updatePetWithForm)
+            post("/pet/{petId}/uploadImage", PetApi::uploadFile)
+        }
+        .routes {
+            delete("/store/order/{orderId}", StoreApi::deleteOrder)
+            get("/store/inventory", StoreApi::getInventory)
+            get("/store/order/{orderId}", StoreApi::getOrderById)
+            post("/store/order", StoreApi::placeOrder)
+        }
+        .routes {
+            post("/user", UserApi::createUser)
+            post("/user/createWithArray", UserApi::createUsersWithArrayInput)
+            post("/user/createWithList", UserApi::createUsersWithListInput)
+            delete("/user/{username}", UserApi::deleteUser)
+            get("/user/{username}", UserApi::getUserByName)
+            get("/user/login", UserApi::loginUser)
+            get("/user/logout", UserApi::logoutUser)
```

Comment:
> Any suggestions on how to do this in the Mustache templates?

sadly no easy way to do it in mustache template

> Otherwise I can check with the Javalin devs is there's a more efficient way to do it.

yes please

>  It could add some complexity to the templates though

for complex logic, better do it in the generator (java class) and create x-something for example in postProcessOperationsWithModels, postProcessModels, etc

Reply:
> yes please

Done, let's see what they suggest: https://github.com/javalin/javalin/issues/2078

Reply:
Fixed in https://github.com/OpenAPITools/openapi-generator/pull/17596/commits/bc1450e8edc4ca545f2bf53d2d215f83d2d7efd7 ✅